### PR TITLE
EZP-26866: Prototype integration of YUI based 1.x components into the Hybrid Symfony app

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-26866
Requirements: https://github.com/ezsystems/requirements/pull/13

# Description

Integration of YUI based 1.x components in an Symfony App as Custom Element.

### CODE REVIEW

File: docs/subitem-udw-in-sf-app.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# Reuse PlatformUI 1.x YUI based components in PlatformUI 2.x
+
+* status: draft
+* author: dp@ez.no
+
+## Idea
+
+Expose some YUI based component as WebComponents. Relying on 1.x code would then
+be a temporary solution and progressively we can get rid of YUI based code while
+still keeping the WebComponents. Other said, the custom element would become a
```
Comment (andrerom) [inline docs/subitem-udw-in-sf-app.md]: >  Other said, the custom element would become a public API to use those components.

Maybe => "Said in other words, the custom element servers as a public api _(hiding implementation details for those consuming them)_."

File: docs/subitem-udw-in-sf-app.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+This component is mostly declarative. It only needs the parent Location id of
+the subitem to render.
+
+Since PlatformUI 1.x is based on the REST API, it only handles REST id ie
+resource URL in the REST API for a given domain object. So in the easiest
+solution would be to pass the *REST Location id* to the component, like:
+
+```xml
+<ez-subitem
+    parent-location-id="/api/ezp/v2/content/locations/1/43/51/59/73/74"
```
Comment (andrerom) [inline docs/subitem-udw-in-sf-app.md]: if this is going to be public api we would need to hide implementation details like this somehow, either on js side without having to do _(with or without improvements on REST API)_, or on api side by for instance expect some form of serialized location object _(ideally without need for twig helper, as in making sure `_toString()` gives the right format or something)_

**Comment by dpobel** [comment]
As far as I'm concerned, the topic "Reuse YUI based components in a Symfony App" is clear :) I wrote a summary in the last commit. @dew326 for doing the same with the Landing Page Editor, I guess you should read that (and the rest of the patch :)). Feel free to ping me if something is unclear.

@yannickroger @StephaneDiot @andrerom (and others) feel free to read it as well to at least get an overview of the topic.

File: docs/subitem-udw-in-sf-app.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+{# Twig template to view a Content / Location #}
+<h1>{{ content.name }}</h1>
+
+<!-- ... -->
+
+{# Here I want to render the subitem of the current Location #}
+{# I just generate a ez-subitem with some attributes tag like any other HTML tag #}
+<ez-subitem
+    parent-location-id="{{ path( 'ezpublish_rest_loadLocation', location ) }}"
+    ></ez-subitem-list>
```
Comment (gggeek) [inline docs/subitem-udw-in-sf-app.md]: close a different tag ?

**Pull request closed without merging**